crates/pallet-domains/src/block_tree.rs

```diff
@@ -0,0 +1,504 @@
+    }
+    Ok(())
+}
+
+/// Import the genesis receipt to the block tree
+pub(crate) fn import_genesis_receipt<T: Config>(
+    domain_id: DomainId,
+    genesis_receipt: ExecutionReceiptOf<T>,
+) {
+    let er_hash = genesis_receipt.hash();
+    let domain_block_height = genesis_receipt.domain_block_height;
+    let domain_block = DomainBlock {
+        execution_receipt: genesis_receipt,
+        operator_ids: sp_std::vec![],
+    };
+    // NOTE: no need to upate the head receipt number as we are using `ValueQuery`
+    BlockTree::<T>::mutate(domain_id, domain_block_height, |er_hashes| {
+        er_hashes.try_insert(er_hash)
+            .expect(
+                "Must not exceed MaxBlockTreeFork as the genesis receipt is the first and only receipt at block #0; qed"
+            );
+    });
+    DomainBlocks::<T>::insert(er_hash, domain_block);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::domain_registry::DomainConfig;
+    use crate::tests::{
+        create_dummy_bundle_with_receipts, create_dummy_receipt, new_test_ext,
+        GenesisStateRootGenerater, ReadRuntimeVersion, Test,
+    };
+    use crate::NextDomainId;
+    use frame_support::dispatch::RawOrigin;
+    use frame_support::traits::{Currency, Hooks};
+    use frame_support::weights::Weight;
+    use frame_support::{assert_err, assert_ok};
+    use frame_system::Pallet as System;
+    use sp_core::H256;
+    use sp_domains::{GenesisReceiptExtension, RuntimeType};
+    use sp_runtime::traits::BlockNumberProvider;
+    use sp_version::RuntimeVersion;
+    use std::sync::Arc;
+
+    fn run_to_block<T: Config>(block_number: T::BlockNumber, parent_hash: T::Hash) {
+        System::<T>::set_block_number(block_number);
+        System::<T>::initialize(&block_number, &parent_hash, &Default::default());
+        <crate::Pallet<T> as Hooks<T::BlockNumber>>::on_initialize(block_number);
+        System::<T>::finalize();
+    }
+
+    #[test]
+    fn test_block_tree() {
+        let creator = 1u64;
+        let operator_id = 1u64;
+        let block_tree_pruning_depth = <Test as Config>::BlockTreePruningDepth::get() as u64;
+
+        let version = RuntimeVersion {
+            spec_name: "test".into(),
+            impl_name: Default::default(),
+            authoring_version: 0,
+            spec_version: 1,
+            impl_version: 1,
+            apis: Default::default(),
+            transaction_version: 1,
+            state_version: 0,
+        };
+
+        let domain_config = DomainConfig {
+            domain_name: b"evm-domain".to_vec(),
+            runtime_id: 0,
+            max_block_size: 1u32,
+            max_block_weight: Weight::from_parts(1, 0),
+            bundle_slot_probability: (1, 1),
+            target_bundles_per_block: 1,
+        };
+
+        let mut ext = new_test_ext();
+        ext.register_extension(sp_core::traits::ReadRuntimeVersionExt::new(
+            ReadRuntimeVersion(version.encode()),
```

Comment:
Okay, I see this is used here. I'm still not sure how block tree tests need that when you can create the required state already.

Either way, would have preferred previous code move in this commit instead

Reply:
> I'm still not sure how block tree tests need that when you can create the required state already.

The test call `Pallet::<Test>::register_domain_runtime` directly.